### PR TITLE
Release 2.0.0

### PR DESCRIPTION
**Pull request by AlexBenny**
### CHANGELOG
- Set lifecycle tracking off by default (#443)
- Bump min Android API to 21+ (#445)
- Fix crash with OkHttp 4.0.0 on API-30 (#446)
- Update README (#369)
- Fix crash on demo app on API 30 (#438)
- Publish to MavenCentral (#436)
- Fix StrictMode alerts (#434)
- Remove protocol from NetworkController and NetworkConnection (#432)
- Store sessions info separately per tracker namespace (#430)
- Fix OkHttpClient sslSocketFactory always overriden (#417)
- Add API reference documentation for configurations and controllers (#428)
- Allow multiple instances of the tracker (#426)
- Create API for v.2.0 (#422)
- Add Nullability annotations to methods and parameters (#420)
- Migrate to AndroidX (#411)
- Bump min Android API to 16+ (#349)

### CODE REVIEW

File: snowplow-demo-app/build.gradle [hunk 1]
```diff
@@ -43,11 +43,11 @@ android {
...
-    implementation "android.arch.lifecycle:extensions:$project.archLifecycleVersion"
+    implementation "androidx.lifecycle:lifecycle-extensions:2.2.0"
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation project(':snowplow-android-tracker')
-    implementation 'com.android.support:customtabs:28.0.0'
-    implementation 'com.android.support:appcompat-v7:28.0.0'
-    implementation 'com.squareup.okhttp3:okhttp:3.4.1'
+    implementation 'androidx.browser:browser:1.3.0'
+    implementation 'androidx.appcompat:appcompat:1.2.0'
+    implementation 'com.squareup.okhttp3:okhttp:4.0.0'
```
Comment (paulboocock) [inline snowplow-demo-app/build.gradle:51]: Should we go up to 4.9.1?

File: snowplow-tracker/build.gradle [hunk 2]
```diff
@@ -76,13 +76,16 @@ android {
 }
 
 dependencies {
-    implementation 'com.android.support:support-annotations:28.0.0'
-    implementation "android.arch.lifecycle:extensions:$project.archLifecycleVersion"
+    implementation 'androidx.annotation:annotation:1.1.0'
+    implementation "androidx.lifecycle:lifecycle-extensions:2.2.0"
     implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation 'com.squareup.okhttp3:okhttp:3.12.12'
+    implementation 'com.squareup.okhttp3:okhttp:4.0.0'
```
Comment (paulboocock) [inline snowplow-tracker/build.gradle:82]: Same question here

File: snowplow-tracker/build.gradle [hunk 3]
```diff
@@ -76,13 +76,16 @@ android {
...
     // test
-    androidTestImplementation 'com.google.android.gms:play-services-analytics:16.0.1'
-    androidTestImplementation 'com.squareup.okhttp3:mockwebserver:3.12.12'
+    testImplementation 'junit:junit:4.13.1'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.2'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+    androidTestImplementation 'com.google.android.gms:play-services-analytics:17.0.0'
+    androidTestImplementation 'com.squareup.okhttp3:mockwebserver:4.0.0'
```
Comment (paulboocock) [inline snowplow-tracker/build.gradle:88]: And here

File: snowplow-tracker/build.gradle [hunk 4]
```diff
@@ -101,60 +104,22 @@ coveralls {
...
-        userOrg = 'snowplow'
-
-        websiteUrl = 'https://github.com/snowplow/snowplow-android-tracker'
-        vcsUrl = 'https://github.com/snowplow/snowplow-android-tracker'
-        issueTrackerUrl = 'https://github.com/snowplow/snowplow-android-tracker/issues'
-
-        licenses = ['Apache-2.0']
-
-        version.name = "$project.version"
+// Deprecated: to fix as explained in https://github.com/vanniktech/gradle-maven-publish-plugin/issues/137
```
Comment (paulboocock) [inline snowplow-tracker/build.gradle:111]: This doesn't need to happen for beta.1 but would be good to switch this over.

**Comment by paulboocock** [comment]
The DemoApp crashes at the moment when using API 30. It looks like this is a OkHttp3 issue: https://github.com/bumptech/glide/issues/4360
Bumping to 4.9 fixed this for me.

Can we also remove these two StrictMode settings? https://github.com/snowplow/snowplow-android-tracker/blob/release/2.0/snowplow-demo-app/src/main/java/com/snowplowanalytics/snowplowtrackerdemo/Demo.java#L98
I get some weird pop ups in the Demo app unless I remove these now (I guess #434 fixes these internally so we don't need them in the App?)

File: snowplow-tracker/build.gradle (hunk 2 above)
Comment (AlexBenny) [reply snowplow-tracker/build.gradle:82]: I would keep this one in the tracker for now. I'm unsure of possible consequences. I wouldn't set the requirements too high and cut out all the ones with a different version.

File: snowplow-tracker/build.gradle (hunk 3 above)
Comment (AlexBenny) [reply snowplow-tracker/build.gradle:88]: same answer above. I'll give a second look to this in the next beta version.

File: snowplow-tracker/build.gradle (hunk 4 above)
Comment (AlexBenny) [reply snowplow-tracker/build.gradle:111]: Yep. I tried but it raised errors so I just put there a reminder.

**Review by paulboocock — APPROVED** [review]

**Review by paulboocock — APPROVED** [review]

**Pull request merged**